.github/actions/build-workspace-docker/action.yml

```diff
@@ -0,0 +1,83 @@
+name: "build ugbio workspace docker"
+description: "general action to build and push ugbio workspace docker"
+
+inputs:
+  workspace-folder:
+    description: "workspace folder under src"
+    required: true
+  docker-repo:
+    description: "docker image repository name"
+    required: true
+  docker-tag:
+    description: "Docker image tag, If omitted will use github short sha"
+    required: false
+  docker-file:
+    description: "Dockerfile path"
+    required: true
+  docker-context:
+    description: "Docker build context"
+    required: true
+    default: "."
+
+runs:
+  using: "composite"
+
+  steps:
+  - uses: benjlevesque/short-sha@v2.2
+    id: short-sha
+    with:
+      length: 7
+
+  - name: set tag
```

Comment:
I'm not sure about this tag logic... I think we should read it from the pyproject.toml version